addons/mail/static/src/messaging/component/thread_viewer/thread_viewer.js

```diff
@@ -112,9 +130,20 @@ class ThreadViewer extends Component {
      * @returns {Object}
      */
     _useStoreSelector(props) {
+        const self = this;
         const threadViewer = this.env.entities.ThreadViewer.get(props.threadViewerLocalId);
         const thread = threadViewer ? threadViewer.thread : undefined;
         const threadCache = threadViewer ? threadViewer.threadCache : undefined;
+        if (threadCache && (threadCache.isLoading || !threadCache.isLoaded)) {
+            if (!this._loaderTimeout) {
+                this._loaderTimeout = setTimeout(() => {
+                    self.state.hasLoader = threadCache.isLoading || !threadCache.isLoaded;
+                    self._loaderTimeout = null;
+                }, 400);
```

Comment:
I would move this logic in entity, because I think it's quite unsafe having asynchronous logic in component, so rule of thumb is usually move to entity in that case.
Also entity thread viewer directly maps a thread viewer component, so it shouldn't be an issue in this case.

Reply:
Would you use another field or handle this with `isLoading` or `isLoaded` ?

Reply:
`isLoading` and `isLoaded` is dependent on thread cache. Here it's about displaying loading or not on thread viewer, which is a bit different.

I think a new attribute should be introduced in thread viewer, and it should have timeout for auto-updating this attribute.

Reply:
Ok that's what I was thinking, I'm working on it, thx